test/parse/TestPythonParser.cpp

```diff
@@ -0,0 +1,101 @@
+    BOOST_REQUIRE(!techs.empty());
+    BOOST_REQUIRE(!tech_categories.empty());
+    BOOST_REQUIRE(!categories_seen.empty());
+
+    for (const auto& cat : {"PRODUCTION_CATEGORY", "GROWTH_CATEGORY"}) {
+        BOOST_REQUIRE_EQUAL(1, categories_seen.count(cat));
+        BOOST_REQUIRE_EQUAL(1, tech_categories.count(cat));
+    }
+
+    const auto cat_it = tech_categories.find("CONSTRUCTION_CATEGORY");
+    BOOST_REQUIRE(tech_categories.end() != cat_it);
+
+    BOOST_REQUIRE_EQUAL("CONSTRUCTION_CATEGORY", cat_it->second->name);
+    BOOST_REQUIRE_EQUAL("construction.png", cat_it->second->graphic);
+    const std::array<unsigned char, 4> test_colour{241, 233, 87, 255};
+    BOOST_REQUIRE(test_colour == cat_it->second->colour);
+
+    {
+        const auto tech_it = techs.get<TechManager::NameIndex>().find("LRN_ALGO_ELEGANCE");
+        BOOST_REQUIRE(techs.get<TechManager::NameIndex>().end() != tech_it);
+        BOOST_REQUIRE_EQUAL("LRN_ALGO_ELEGANCE", (*tech_it)->Name());
+        BOOST_REQUIRE_EQUAL("LRN_ALGO_ELEGANCE_DESC", (*tech_it)->Description());
+        BOOST_REQUIRE_EQUAL("RESEARCH_SHORT_DESC", (*tech_it)->ShortDescription());
+        BOOST_REQUIRE_EQUAL("LEARNING_CATEGORY", (*tech_it)->Category());
+        BOOST_REQUIRE_EQUAL(true, (*tech_it)->Researchable());
+        BOOST_REQUIRE_EQUAL("icons/tech/algorithmic_elegance.png", (*tech_it)->Graphic());
+
+        BOOST_REQUIRE_CLOSE(0.0, (*tech_it)->ResearchCost(ALL_EMPIRES), 0.1);
+        BOOST_REQUIRE_EQUAL(3, (*tech_it)->ResearchTime(ALL_EMPIRES));
+        BOOST_REQUIRE_EQUAL(0, (*tech_it)->Effects().size());
+        BOOST_REQUIRE_EQUAL(0, (*tech_it)->UnlockedTechs().size());
+        BOOST_REQUIRE_EQUAL(0, (*tech_it)->Prerequisites().size());
+
+        const auto& tech_tags = (*tech_it)->Tags();
+        BOOST_REQUIRE_EQUAL(1, tech_tags.size());
+        BOOST_REQUIRE_EQUAL(1, tech_tags.count("PEDIA_LEARNING_CATEGORY"));
+
+        const auto& tech_items = (*tech_it)->UnlockedItems();
+        BOOST_REQUIRE_EQUAL(1, tech_items.size());
+        BOOST_REQUIRE_EQUAL(UnlockableItemType::UIT_POLICY, tech_items[0].type);
+        BOOST_REQUIRE_EQUAL("PLC_ALGORITHMIC_RESEARCH", tech_items[0].name);
+    }
+
+    {
+        const auto tech_it = techs.get<TechManager::NameIndex>().find("CON_ORBITAL_HAB");
+        BOOST_REQUIRE(techs.get<TechManager::NameIndex>().end() != tech_it);
+
+        BOOST_REQUIRE_EQUAL(1, (*tech_it)->Effects().size());
+        const auto& tech_effect_gr = (*tech_it)->Effects()[0];
+        BOOST_REQUIRE_EQUAL("", tech_effect_gr->StackingGroup());
+        BOOST_REQUIRE_EQUAL("", tech_effect_gr->GetDescription());
+        BOOST_REQUIRE_EQUAL("", tech_effect_gr->AccountingLabel());
+        BOOST_REQUIRE_EQUAL(17, tech_effect_gr->Priority());
+
+        BOOST_REQUIRE_EQUAL(1, (*tech_it)->Prerequisites().size());
+        BOOST_REQUIRE_EQUAL(1, (*tech_it)->Prerequisites().count("PRO_MICROGRAV_MAN"));
+    }
+
+    // test it last
```

Comment:
In general, it is a bad practice to do multiple tests at once. 

In Python, I can solve it with fixtures:  create a session scope fixture (execute once, reuse in multiple tests).
Quick googling shows that it is possible to create only test level fixtures. 

You mention that the parser is fast, so probably we can run it on each test.

Reply:
These tests are for old parser and it took 18s to parse techs. I will switch to python parser after I write enough test to cover all accessible fields.